src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/PublishingCompletionService.cs

```diff
@@ -117,7 +116,8 @@ await releaseIdsToUpdate
 
         await directlyRelatedPublicationIds
             .ToAsyncEnumerable()
-            .ForEachAwaitAsync(_publicationRepository.UpdateLatestPublishedRelease);
+            .ForEachAwaitAsync(async publicationId => 
+                await UpdateLatestPublishedRelease(publicationId));
```

Comment:
We use `ForEachAwaitAsync` so that we await each call before beginning the next. EF Core doesn't support database calls being made on the same context instance in parallel.

Btw, I didn't make much of a change here except inlining the `UpdateLatestPublishedRelease` mehod into this service from its previous location in `PublicationRepository`.

It was previously:

```csharp
        await directlyRelatedPublicationIds
            .ToAsyncEnumerable()
            .ForEachAwaitAsync(_publicationRepository.UpdateLatestPublishedRelease);
```

and I changed it to:

```csharp
        await directlyRelatedPublicationIds
            .ToAsyncEnumerable()
            .ForEachAwaitAsync(UpdateLatestPublishedRelease);
```

except that I didn't write it with the method group, I used the lambda expression with the `publication` variable:

```csharp
        await directlyRelatedPublicationIds
            .ToAsyncEnumerable()
            .ForEachAwaitAsync(async publicationId => 
                await UpdateLatestPublishedRelease(publicationId));
```

I will rewrite it to use the method group.